jni/src/jni_util.cpp

```diff
@@ -294,6 +294,40 @@ void knn_jni::JNIUtil::Convert2dJavaObjectArrayAndStoreToByteVector(JNIEnv *env,
     env->DeleteLocalRef(array2dJ);
 }
 
+void knn_jni::JNIUtil::Convert2dJavaObjectArrayAndStoreToByteVector(JNIEnv *env, jobjectArray array2dJ,
+                                                                     int dim, std::vector<int8_t> *vect) {
+
+    if (array2dJ == nullptr) {
+        throw std::runtime_error("Array cannot be null");
+    }
+
+    int numVectors = env->GetArrayLength(array2dJ);
+    this->HasExceptionInStack(env);
+
+    for (int i = 0; i < numVectors; ++i) {
+        auto vectorArray = (jbyteArray)env->GetObjectArrayElement(array2dJ, i);
+        this->HasExceptionInStack(env, "Unable to get object array element");
+
+        if (dim != env->GetArrayLength(vectorArray)) {
+            env->DeleteLocalRef(array2dJ);
+            throw std::runtime_error("Dimension of vectors is inconsistent");
+        }
+
+        int8_t* vector = reinterpret_cast<int8_t*>(env->GetByteArrayElements(vectorArray, nullptr));
+        if (vector == nullptr) {
+            this->HasExceptionInStack(env);
+            throw std::runtime_error("Unable to get byte array elements");
+        }
+
+        for(int j = 0; j < dim; ++j) {
```

Comment:
We can use standard library given 

like 

vect->insert(vect->end(), vector, vector + dim);